portfolio/src/main/webapp/script.js

```diff
@@ -29,3 +29,11 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+function testFetch() {
+    console.log("Test fetch from data")
+
+    fetch('/data').then(response => response.text()).then((data) => {
+        document.getElementById('greeting-container').innerHTML = data;
```

Comment:
Watch out! While this is safe as written, avoid "innerHTML" in general as it can lead to major security vulnerabilities. (Prefer textContent/innerText)

See
https://xss-game.appspot.com/

Reply:
(Note this would break returning <h1> as part of the response!) Just wanted to make sure you were aware of the implications

Reply:
This is interesting! Thanks for the game link :)